### ISSUE TITLE
Deploying/Deactivating turrets duplicates ammo and materializes magazines.

### ISSUE THREAD

**Issue by John-Candlebury** [CleverRaven/Cataclysm-DDA#56603]
### Describe the bug

When deploying a robot turret with ammo belts, you'll transfer the ammo to the monster inventory, but neither the ammo nor the belt will be removed from the inventory. If you then deactivate the turret youll also recieve a copy of all the ammo you didnt actually put there.

Additionally if the turret is killed, the duplicated ammo will be contained in a brand new ammo belt/compatible magazine.

### Steps to reproduce

1. Spawn a turret and a compatible ammo belt.
2. Deploy the turret, you'll load ammo into it, but it wont dissapear from your inventory.
3. Deactivate the turret, ammo is duplicated.
4. Repeat step 2 but now kill the turret instead of deactivating it. The dropped ammo is contained in a new magazine.

### Expected behavior

No Duplications

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.1586 (21H2)
- Game Version: 0.F-7149-gbeef37a4c5-dirty [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
]

**Comment by Night-Pryanik** [comment]
Could you give an example of deployable turret with ammo belts? AFAIK there are no such turrets in vanilla.

For ordinary vanilla 9x19mm turret all is working as it should: game correctly removes ammo from the inventory on deploying of turret and gives them back on deactivating it.

**Comment by John-Candlebury** [comment]
Yes of course this Issue was tested and reproduced with a 'M249 autonomous CROWS II' which can be stored and redeployed in game

**Comment by Night-Pryanik** [comment]
I still can't reproduce this issue on 0.F-12274. 

I have a M249 CROWS II and an ammo belt loaded with 500 5.56 M855.
![изображение](https://user-images.githubusercontent.com/11132525/201478147-9ce371cd-e574-4109-8743-8e2babe030f4.png)

I deployed the turret, but it didn't take ammo from the ammo belt:
![изображение](https://user-images.githubusercontent.com/11132525/201478214-0ec589a5-61eb-42d1-97cf-edded4a2fae8.png)
![изображение](https://user-images.githubusercontent.com/11132525/201478227-7499c519-2c41-4e3f-bfe8-a880b80524d7.png)

The turret has taken loose ammo from the inventory:
![изображение](https://user-images.githubusercontent.com/11132525/201478342-67d676f7-f3e5-4306-b8a0-eab0575b30df.png)
![изображение](https://user-images.githubusercontent.com/11132525/201478349-1d9ac776-c963-4031-b1bd-e835aa6d7d3e.png)

Deactivating the non-loaded turret didn't give me any ammo (obviously)
